solr/core/src/java/org/apache/solr/handler/admin/api/SnapshotBackupAPI.java

```diff
@@ -0,0 +1,174 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.solr.handler.admin.api;
+
+import static org.apache.solr.client.solrj.impl.BinaryResponseParser.BINARY_CONTENT_TYPE_V2;
+import static org.apache.solr.handler.ReplicationHandler.ERR_STATUS;
+import static org.apache.solr.security.PermissionNameProvider.Name.CORE_EDIT_PERM;
+
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.media.Schema;
+import io.swagger.v3.oas.annotations.parameters.RequestBody;
+import java.lang.invoke.MethodHandles;
+import java.util.function.Consumer;
+import javax.inject.Inject;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import org.apache.solr.api.JerseyResource;
+import org.apache.solr.client.api.model.SolrJerseyResponse;
+import org.apache.solr.common.SolrException;
+import org.apache.solr.common.annotation.JsonProperty;
+import org.apache.solr.common.util.NamedList;
+import org.apache.solr.core.SolrCore;
+import org.apache.solr.handler.ReplicationHandler;
+import org.apache.solr.handler.ReplicationHandler.ReplicationHandlerConfig;
+import org.apache.solr.jersey.JacksonReflectMapWriter;
+import org.apache.solr.jersey.PermissionName;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** V2 endpoint for Backup API used for User-Managed clusters and Single-Node Installation. */
+@Path("/cores/{coreName}/replication/backups")
+public class SnapshotBackupAPI extends JerseyResource {
```

Comment:
[+1] Thanks for updating the path here to match what we lined up in the spreadsheet!

[-0] There are a LOT of different backup or snapshot adjacent APIs: `/admin/collections?action=BACKUP`, `/admin/cores?action=BACKUP`, `/admin/collections?action=CREATESNAPSHOT`, `/admin/cores?action=CREATESNAPSHOT`, `/replication?cmd=backup`, etc.  With so many similar APIs, it's really hard to find names to differentiate them all.  (Tbh I'm not even sure what the differences are between the ReplicationHandle backup functionality and what's exposed via `/admin/collections` and `/admin/cores`.)

But if you've got any sparks of inspiration there and can come up with a name that helps distinguish this API from its close cousins, we should definitely use that here.

(As a larger question, I think we should re-evaluate which of these functionalities is worth keeping around in 10.0.  So maybe this disambiguation question is only a short-term concern.  But still 🤷 )

Reply:
hmm. I see `snapshot` as a term used on the replication handler. maybe this would work to not overload the `backup` term?